### PR TITLE
case task component

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-3626
https://tools.hmcts.net/jira/browse/EUI-3778
https://tools.hmcts.net/jira/browse/EUI-3779

### Change description ###
Display Content Task Tab Components

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/cases/components/case-task/case-task.component.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+<p class="govuk-body">
+    <strong>{{task.task_title}}</strong>
+    <span class="hmcts-badge_right"></span>
+</p>
+<dl class="govuk-summary-list govuk-summary-list--no-border">
+    <div class="govuk-summary-list__row">
+        <dt class="govuk-summary-list__key">
+            <span class="row-padding">Task created</span>
+        </dt>
+        <dd class="govuk-summary-list__value">2 September 2021</dd>
```
Comment (kuda1992) [inline src/cases/components/case-task/case-task.component.html:11]: Are we hardcoding the date?

**Review by kuda1992 — CHANGES_REQUESTED** [review]
Looks fine to me apart from the hard-coded date

File: src/cases/components/case-task/case-task.component.html (hunk 1 above)
Comment (udaydenduluri33) [reply src/cases/components/case-task/case-task.component.html:11]: good point. Let me change it

Comment (udaydenduluri33) [reply src/cases/components/case-task/case-task.component.html]: Done

**Review by kuda1992 — APPROVED** [review]
Thanks @udaydenduluri33 for making the changes. All good to me now.

File: src/cases/components/case-task/case-task.component.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    return false;
+  }
+
+  public doesUserHaveOwnAndExecute(task: Task): boolean {
+    return task.permissions.includes('Own') && task.permissions.includes('Execute');
+  }
+
+  public getManageOptions(task: Task): Array<{text: string, path: string}> {
+    if (!task.assignee) {
+      if (task.permissions.length === 0 || (task.permissions.length === 1 && task.permissions.includes('Manage'))) {
```
Comment (rayliangatos) [inline src/cases/components/case-task/case-task.component.ts:40]: 'Manage' can be replaced with TaskPermission.MANAGE

File: src/cases/components/case-task/case-task.component.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    if (userInfoStr) {
+      const userInfo: UserInfo = JSON.parse(userInfoStr);
+      const userId = userInfo.id ? userInfo.id : userInfo.uid;
+      return task.assignee && task.assignee === userId;
+    }
+    return false;
+  }
+
+  public doesUserHaveOwnAndExecute(task: Task): boolean {
+    return task.permissions.includes('Own') && task.permissions.includes('Execute');
```
Comment (rayliangatos) [inline src/cases/components/case-task/case-task.component.ts:35]: 'Execute' can be replaced with TaskPermission.EXECUTE

Comment (rayliangatos) [reply src/cases/components/case-task/case-task.component.ts:35]: 'Own' can be replaced with TaskPermission.OWN

File: src/cases/components/case-task/case-task.component.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      }
+    }
+
+    if (this.isTaskAssignedToCurrentUser(task)) {
+      return [
+        {text: 'Reassign task', path: `/work/${task.id}/reassign`},
+        {text: 'Unassign task', path: `/work/${task.id}/unclaim`}
+      ];
+    } else {
+      if (task.permissions.includes('Execute') && task.permissions.includes('Manage')) {
```
Comment (rayliangatos) [inline src/cases/components/case-task/case-task.component.ts:53]: ditto

File: src/cases/components/case-task/case-task.component.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        return [
+          {text: 'Assign to me', path: ''},
+          {text: 'Reassign task', path: `/work/${task.id}/reassign`},
+          {text: 'Unassign task', path: `/work/${task.id}/unclaim`}
+        ];
+      } else if (task.permissions.includes('Manage')) {
```
Comment (rayliangatos) [inline src/cases/components/case-task/case-task.component.ts:59]: ditto

Comment (udaydenduluri33) [reply src/cases/components/case-task/case-task.component.ts]: Done

File: src/cases/components/case-task/case-task.component.ts (hunk 4 above)
Comment (udaydenduluri33) [reply src/cases/components/case-task/case-task.component.ts]: Done

File: src/cases/components/case-task/case-task.component.ts (hunk 3 above)
Comment (udaydenduluri33) [reply src/cases/components/case-task/case-task.component.ts]: Done

File: src/cases/components/case-task/case-task.component.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/cases/components/case-task/case-task.component.ts]: Done

File: api/workAllocation2/taskService.mock.ts [hunk 6]
```diff
@@ -160,10 +162,9 @@ export const init = () => {
 
   mock.onGet(getTasksByCaseIdUrl).reply(config => {
     // return an array in the form of [status, data, headers]
-    const caseId = config.url.match(/[a-fA-F0-9]{16}/);
-    let tasks = ALL_TASKS.tasks.filter(task => task.case_id === caseId[0]);
+    const tasks = [...ASSIGNED_CASE_TASKS.tasks, ...UNASSIGNED_CASE_TASKS.tasks];
     // note: added next line to ensure results returned in all circumstances
-    tasks = tasks === [] ? tasks : JUDICIAL_AVAILABLE_TASKS.tasks;
+    // tasks = tasks === [] ? tasks : JUDICIAL_AVAILABLE_TASKS.tasks;
```
[the remark this answers is not in the document]
Comment (udaydenduluri33) [reply api/workAllocation2/taskService.mock.ts]: Done

File: api/workAllocation2/constants/mock.data.ts [hunk 7]
```diff
@@ -1771,9 +1771,153 @@ export const JUDICIAL_WORKERS = [
...
+      location_id: '512401',
+      case_id: '1620409659381330',
+      case_category: 'refusalOfHumanRights',
+      case_name: 'Francis Gigs',
+      permissions: ['Own', 'Execute', 'Manage'],
+    },
+    {
+      assignee: 'bce92ab6-0023-46e3-886b-7f61384df090',
+      assigneeName: 'Judicial User 10',
+      id: '0d22d838-b25a-11eb-a18c-f2d58a9b7bc4',
```
[the remark this answers is not in the document]
Comment (udaydenduluri33) [reply api/workAllocation2/constants/mock.data.ts]: Agreed, Changed it now

**Review by rayliangatos — APPROVED** [review]

**Pull request merged**